pkg/deploy/defaults.go

```diff
@@ -76,6 +79,12 @@ const (
 	DefaultSecurityContextFsGroup   = "1724"
 	DefaultSecurityContextRunAsUser = "1724"
 
+	DefaultServerExposureStrategy              = "multi-host"
+	DefaultKubernetesSingleHostExposureType    = "native"
+	DefaultOpenShiftSingleHostExposureType     = "gateway"
+	DefaultSingleHostGatewayImage              = "docker.io/traefik:v2.2.8"
```

Comment:
I assume that the docker image references here should also be configurable through `RELATED_IMAGE_xxx` properties as done for the other installation components, to enable the support of downstreaming and digest management.
cc @tolusha

Reply:
I think it already is
```
# The image used for the gateway in the single host mode. Defaults to docker.io/traefik:v2.2.8.
singleHostGatewayImage: ''
# The image used for the gateway sidecar that provides configuration to the gateway.
# Default to quay.io/che-incubator/configbump:latest
singleHostGatewayConfigSidecarImage: ''

Reply:
I'm not saying _configurable from the `CheCluster` custom resource_. This is clearly implemented.
I'm speaking of the ability to configure the default value itself (when nothing is defined in the CR) with RELATED_IMAGES_XXX environment variables passed to the operator deployment, by the OLM ClusterServiceVersion manifest. This is a requirement for downstreaming and for being compatible with Openshift digest management and disconnected OLM installation mode.

To be more precise:
- I don't see a GO variable for each of the new docker images [here](https://github.com/eclipse/che-operator/blob/c4484c0c56dd5a002acf48b0a2cf5d5f20bc13b5/pkg/deploy/defaults.go#L30), but default values of the new docker images are provided as constants.
- I don't see the default values of the new docker images initialized based on environment variables in the [`InitDefaultsFromEnv`](https://github.com/eclipse/che-operator/blob/c4484c0c56dd5a002acf48b0a2cf5d5f20bc13b5/pkg/deploy/defaults.go#L108) method for example.

Reply:
Thanks for the suggestion. I believe I added the necessary code but I am a little bit ignorant about how OLM and chectl interact. With this change, the operator is not functional when deployed using chectl, because the operator deployment lacks the above env vars.

Reply:
Thanks !